sql/core/src/main/java/org/apache/spark/sql/sources/v2/reader/ContinuousDataReader.java

```diff
@@ -0,0 +1,29 @@
+package org.apache.spark.sql.sources.v2.reader;
+
+import org.apache.spark.sql.sources.v2.reader.PartitionOffset;
+
+import java.io.IOException;
+
+/**
+ * A variation on {@link DataReader} for use with streaming in continuous processing mode.
+ */
+public interface ContinuousDataReader<T> extends DataReader<T> {
+    /**
+     * Proceed to next record, returning false only if the read is interrupted.
+     *
+     * @throws IOException if failure happens during disk/network IO like reading files.
+     */
+    boolean next() throws IOException;
+
+    /**
+     * Return the current record. This method should return same value until `next` is called.
+     */
+    T get();
```

Comment:
I think we don't need to repeat these 2 methods here, as you already inherit from `DataReader`